dfastmi/batch/AnalyserDflowfm.py

```diff
@@ -0,0 +1,376 @@
+                                        filenames: Dict[Any, Tuple[str,str]],
+                                        ucrit : float,
+                                        n_fields : int,
+                                        dxi : numpy.ndarray,
+                                        dyi : numpy.ndarray,
+                                        iface : numpy.ndarray
+                                        ) -> Tuple[bool, numpy.ndarray]:
+        dzq = [None] * len(discharges)
+        for i in range(3):
+                if not missing_data and discharges[i] is not None:
+                    dzq[i] = self._get_values_fm(discharges[i], ucrit, filenames[i], n_fields, dxi, dyi, iface)
+                    if dzq[i] is None:
+                        missing_data = True
+                else:
+                    dzq[i] = 0
+        return missing_data, dzq
+    
+    def _get_dzq_based_on_conditions_keys(self,
+                                          missing_data : bool,
+                                          discharges : Vector,
+                                          filenames : Dict[Any, Tuple[str,str]],
+                                          tide_bc : Tuple[str, ...],
+                                          rsigma : Vector,
+                                          ucrit : float,
+                                          n_fields : int,
+                                          dxi : numpy.ndarray,
+                                          dyi : numpy.ndarray,
+                                          iface : numpy.ndarray
+                                          ) -> Tuple[bool, numpy.ndarray]:
+        dzq = [None] * len(discharges)
+        for i in range(len(discharges)):
+            if not missing_data and discharges[i] is not None:
+                key, q, t = self._get_condition_key(discharges, tide_bc, i)
+                if rsigma[i] == 1:
+                    # no celerity, so ignore field
+                    dzq[i] = 0
+                elif key in filenames.keys():
+                    if t:
+                        n_fields_request = n_fields
+                    else:
+                        n_fields_request = 1
+                    dzq[i] = self._get_values_fm(q, ucrit, filenames[key], n_fields_request, dxi, dyi, iface)
+                else:
+                    self._logger.report_missing_calculation_dzq_values(q, t)
+                    missing_data = True
+            else:
+                dzq[i] = 0
+        return missing_data, dzq
+    
+    def _get_condition_key(self, discharges : Vector, tide_bc : Tuple[str, ...], i : int):
+        q = discharges[i]
+        if self._needs_tide:
+            t = tide_bc[i]
+            key = (q,t)
+        else:
+            t = None
+            key = q
+        return key,q,t
+
+    def _get_values_fm(
+        self,
+        q: float,
+        ucrit: float,
+        filenames: Tuple[str, str],
```

Comment:
Can you refactor this by per file?

Reply:
I had not really looked much into refactoring this method.
Do we want to do that in this issue, or should I make a followup?